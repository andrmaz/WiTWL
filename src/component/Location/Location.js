import React from 'react'
import './Location.scss';

const Location = ({ weather }) => {
    const { location } = weather;
    const country = location?.country ? location.country : '';
    const localtime = location?.localtime ? location.localtime : '';
    const name = location?.name ? location.name : '';
    const region = location?.region ? location.region : '';
    const tz = location?.tz_id ? location.tz_id : '';
    return (
        <aside className='box2'>
            <article className='wrapper'>
                <h2>Location</h2>
                <p>{`Country: ${country}`}</p>
                <p>{`Local Time: ${localtime}`}</p>
                <p>{`City: ${name}`}</p>
                <p>{`Region: ${region}`}</p>
                <p>{`Time Zone: ${tz}`}</p>
            </article>
        </aside>
    )
}

Location.propTypes = {
    weather(props, propName, componentName){
        if (typeof props[propName] !== 'object'){
            return new Error(
                `Hey, the component ${componentName} 
                needs the prop ${propName} to be an object,
                but you passed a ${typeof props[propName]}`
            )
        }
    },
}

export default Location
