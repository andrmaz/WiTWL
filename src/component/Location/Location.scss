.box2 {

    grid-row-start: 2;
    grid-row-end: 3;
    grid-column-start: 1;
    grid-column-end: 2;

    .wrapper {
        height: 100%;
        width: 60%;
        margin: auto;
        display: flex;
        justify-content: space-evenly;
        align-items: flex-start;
        flex-flow: column;
    }
}
