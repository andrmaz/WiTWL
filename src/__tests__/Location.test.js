import React from 'react'
import { shallow } from 'enzyme'
import Location from '../component/Location/Location'

let wrapper 
const mockWeather = {
    location: {
        country: '',
        localtime: '',
        name: '',
        region: '',
        tz_id: ''
    }
}

describe('Location Component', () => {
    beforeEach(() => {
        expect.hasAssertions()
        wrapper = shallow(<Location weather={mockWeather} />)
    })
    it('should render component correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })
    it('should throw error on failing props validation', () => {
        wrapper = shallow(<Location weather={''} />)
        const fn = () => { throw new Error() }
        expect(fn).toThrow()
    })
})