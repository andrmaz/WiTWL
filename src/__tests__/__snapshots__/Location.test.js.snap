// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Location Component should render component correctly 1`] = `
<aside
  className="box2"
>
  <article
    className="wrapper"
  >
    <h2>
      Location
    </h2>
    <p>
      Country: 
    </p>
    <p>
      Local Time: 
    </p>
    <p>
      City: 
    </p>
    <p>
      Region: 
    </p>
    <p>
      Time Zone: 
    </p>
  </article>
</aside>
`;
